python/uv/private/uv.bzl

```diff
@@ -22,32 +22,177 @@ load(":uv_repositories.bzl", "uv_repositories")
 
 _DOC = """\
 A module extension for working with uv.
+
+Use it in your own setup by:
+```starlark
+uv = use_extension(
+    "@rules_python//python/uv:uv.bzl",
+    "uv",
+    dev_dependency = True,
+)
+uv.toolchain(
+    name = "uv_toolchains",
+    version = "0.5.24",
+)
+use_repo(uv, "uv_toolchains")
+
+register_toolchains(
+    "@uv_toolchains//:all",
+    dev_dependency = True,
+)
+```
+
+Since this is only for locking the requirements files, it should be always
+marked as a `dev_dependency`.
 """
 
+_DIST_MANIFEST_JSON = "dist-manifest.json"
+_DEFAULT_BASE_URL = "https://github.com/astral-sh/uv/releases/download"
+
+config = tag_class(
+    doc = "Configure where the binaries are going to be downloaded from.",
+    attrs = {
+        "base_url": attr.string(
+            doc = "Base URL to download metadata about the binaries and the binaries themselves.",
+            default = _DEFAULT_BASE_URL,
+        ),
+    },
+)
+
+platform = tag_class(
```

Comment:
Something that isn't clear from the doc here is how the word "for" is operating.

Does this define the platforms uv will run on, or the platform that uv will generate a lockfile for? I'm assuming the former.